### PR TITLE
Use list initialization for structs declared in MultiplayerCommon.h

### PR DESCRIPTION
**Pull request by adrianbroher**
This PR:

* Removes the user-declared default constructor for the `SaveGameEmpireData`, `PlayerInfo`, `ServerSaveGameData`, `PlayerSaveHeaderData`, PlayerSetupData` and `ChatHistoryEntity` structs, using default intializers for member variables instead.
* Removes the user-provided constructors for  the `SaveGameEmpireData`, `PlayerInfo`, `ServerSaveGameData`, `PlayerSaveHeaderData`, PlayerSetupData` and `ChatHistoryEntity` structs, using aggregate initialization instead.
* Removes member variables comments for obvious facts.

### CODE REVIEW

File: util/MultiplayerCommon.h [hunk 1]
```diff
@@ -228,31 +184,17 @@ struct FO_COMMON_API ServerSaveGameData {
...
-    std::string             m_starting_species_name;///< name of the species with which the player starts when starting a new game
-    int                     m_save_game_empire_id;  ///< when loading a game, the ID of the empire that this player will control
-    Networking::ClientType  m_client_type;          ///< is this player an AI, human player or...?
-    bool                    m_player_ready;         ///< if player ready to play.
-    bool                    m_authenticated;        ///< if player was authenticated
-    int                     m_starting_team;        ///< team id or -1 if no team.
+    std::string             m_player_name;
+    int                     m_player_id = Networking::INVALID_PLAYER_ID;
+    std::string             m_empire_name;
+    GG::Clr                 m_empire_color = GG::Clr(0, 0, 0, 0);
```
Comment (geoffthemedio) [inline util/MultiplayerCommon.h]: `CLR_ZERO` not accessible?

File: util/MultiplayerCommon.h [hunk 2]
```diff
@@ -228,31 +184,17 @@ struct FO_COMMON_API ServerSaveGameData {
...
+    std::string             m_starting_species_name;
+    //! When loading a game, the ID of the empire that this player will control
+    int                     m_save_game_empire_id = ALL_EMPIRES;
+    Networking::ClientType  m_client_type = Networking::INVALID_CLIENT_TYPE;
+    bool                    m_player_ready;
```
Comment (geoffthemedio) [inline util/MultiplayerCommon.h]: explicitly setting to `false` would seem clearer?

File: util/MultiplayerCommon.h (hunk 1 above)
Comment (adrianbroher) [reply util/MultiplayerCommon.h]: Haven't considered it at the time of writing this change.

File: util/MultiplayerCommon.h (hunk 2 above)
Comment (adrianbroher) [reply util/MultiplayerCommon.h]: True, same goes for `m_authenticated`.

Comment (adrianbroher) [reply util/MultiplayerCommon.h]: Fixed in commit 24830a3fa6965755671094d0ad34810ea51c2d02

File: util/MultiplayerCommon.h (hunk 1 above)
Comment (adrianbroher) [reply util/MultiplayerCommon.h]: Fixed in commit 24830a3fa6965755671094d0ad34810ea51c2d02

**Pull request merged**